### PR TITLE
YLP-1133 get confirm for wiper

### PR DESCRIPTION
**Pull request by yanngit**

### CODE REVIEW

File: clients/hydrophone/hydrophone_client.go [hunk 1]
```diff
@@ -77,23 +79,46 @@ func (client *Client) getHost() (*url.URL, error) {
...
+	} else if len(res) > 1 {
+		return nil, fmt.Errorf("more than one signup found for %s", userID)
+	} else if len(res) == 1 {
+		return &res[0], err
+	} else {
+		return nil, nil
+	}
+}
+
+func (client *Client) GetPendingInvitOrSignup(userID string, authToken string, confirmType models.Type) ([]models.Confirmation, error) {
```
Comment (ydrisR) [inline clients/hydrophone/hydrophone_client.go]:
```suggestion
func (client *Client) GetPendingInviteOrSignup(userID string, authToken string, confirmType models.Type) ([]models.Confirmation, error) {
```

File: clients/hydrophone/hydrophone_client.go [hunk 2]
```diff
@@ -77,23 +79,46 @@ func (client *Client) getHost() (*url.URL, error) {
...
+	} else {
+		host.Path = path.Join(host.Path, "invite", userID)
+	}
 	req, _ := http.NewRequest("GET", host.String(), nil)
 	req.Header.Add("x-tidepool-session-token", authToken)
 
 	res, err := client.httpClient.Do(req)
 	if err != nil {
-		return nil, errors.Wrap(err, "Failure to get pending invites")
+		return nil, errors.Wrap(err, "Failure to get pending signup")
```
Comment (ydrisR) [inline clients/hydrophone/hydrophone_client.go]: signup OR invites ;)

File: clients/hydrophone/hydrophone_client.go [hunk 3]
```diff
@@ -114,3 +139,30 @@ func (client *Client) GetPendingInvitations(userID string, authToken string) ([]
...
 	}
 }
+
+func (client *Client) CancelSignup(userID string, authToken string) error {
+	host, err := client.getHost()
+	if err != nil {
+		return errors.New("No known hydrophone hosts")
+	}
+
+	host.Path = path.Join(host.Path, "signup", userID)
```
Comment (ydrisR) [inline clients/hydrophone/hydrophone_client.go]:
```suggestion
	host.Path = path.Join(host.Path, "dismiss/signup", userID)
```

**Review by ydrisR — CHANGES_REQUESTED** [review]
One typo, and one real error on client method `CancelSignup`.
And by the way there is no UT for this new method ;)

File: clients/hydrophone/hydrophone_client.go (hunk 1 above)
Comment (yanngit) [reply clients/hydrophone/hydrophone_client.go]: fixed

File: clients/hydrophone/hydrophone_client.go (hunk 2 above)
Comment (yanngit) [reply clients/hydrophone/hydrophone_client.go]: confirm :)

File: clients/hydrophone/hydrophone_client.go (hunk 3 above)
Comment (yanngit) [reply clients/hydrophone/hydrophone_client.go]: No, dismiss is a different API ... a PUT on signup is doing a cancel ;) weird but it's the API today ...

**Review by ydrisR — APPROVED** [review]

**Pull request merged**